src/lib/serviceRoutines/postQueryContext.cpp

```diff
@@ -487,17 +527,20 @@ std::string postQueryContext
   // In API version 2, this has changed completely. Here, the total count of local entities is returned
   // if the URI parameter 'count' is set to 'true', and it is returned in the HTTP header Fiware-Total-Count.
   //
-  if ((ciP->apiVersion == V2) && (ciP->uriParamOptions["count"]))
+  std::string georel  =  ciP->uriParam["georel"];
+  if ((ciP->apiVersion == V2))
   {
-    countP = &count;
+    if (!((ciP->inMimeType == JSON && strstr(ciP->payload, "near")) || strstr(georel.c_str(), "near")))
+    {
+      countP = &count;
+    }
```

Comment:
Could you elaborate on the changes done in this part?

I don't get the part related with geolocation ("georel", "near", etc), as it seems geolocation hasn't nothing to do with issue #4149...

Reply:
Without the above geolocation check, all the FTs related with geolocation attributes ("georel", "near") were failing for e.g. FT `1038_ngsiv2_geo_location_entities`, `1177_geometry_and_coords` and `4114_geojson_feature_support`.
So we have added a `georel` related check here for those FTs.

Reply:
I'm trying to "decode" the logical expression... it's hard:

* `(!((ciP->inMimeType == JSON && strstr(ciP->payload, "near")) || strstr(georel.c_str(), "near")))`
    *  `ciP->inMimeType == JSON && strstr(ciP->payload, "near"`: the mime type in the request is JSON and the string `near` is in the payload of the request
    *  `strstr(georel.c_str(), "near")`: the georeal (assuming georeal has been specified) is `near`

So, in sum

> IF NOT (("the mime type in the request is JSON and the string `near` is in the payload of the request") OR "the georeal (assuming georeal has been specified) is `near`")

I don't get why this is needed to avoid some test to fail.  What kind of fail did you get in those FT? Could you provide the diff in this tests (preveribly in a graphical way, e.g. [meld](http://meldmerge.org)?

I'd like to understand this in deep before applying the "guard" you propose...

Reply:
If we are not applying the above geolocation check then below error is occurring on FT.

 I have attach the screenshot of `.out` file for failed FT `test/functionalTest/cases/4114_geojson_feature_support/geojson_feature_support.test`

**Functional Test case:**
![Orion_4114_FT](https://github.com/telefonicaid/fiware-orion/assets/55977298/9b1e0310-a16f-4b3b-b814-6a7eda7f2ae9)

**Output of FT:**
![output_of_failing_FT_without_geolocation_check](https://github.com/telefonicaid/fiware-orion/assets/55977298/d035316f-c2f0-4f7b-ae12-e05f1b734476)



Earlier we have used a simple check at postQueryContext.cpp (L#533) as below:
```
  if (ciP->apiVersion == V2)
  {
     countP = &count;
 }

```
But due to some FTs were failing related with geolocation attribute, so we have added an extra check (at L#536) as below i.e. related with geolocation attributes so that our code changes does not affect the geolocation attribute related APIs:
```
  std::string georel  =  ciP->uriParam["georel"];
  if ((ciP->apiVersion == V2))
  {
    if (!((ciP->inMimeType == JSON && strstr(ciP->payload, "near")) || strstr(georel.c_str(), "near")))
    {
      countP = &count;
    }
  }

```

Reply:
That PR is https://github.com/telefonicaid/fiware-orion/pull/4341

Reply:
I have tested to revert to the original code, this way:

```diff
diff --git a/src/lib/serviceRoutines/postQueryContext.cpp b/src/lib/serviceRoutines/postQueryContext.cpp
index a2952d7b9..64201e799 100644
--- a/src/lib/serviceRoutines/postQueryContext.cpp
+++ b/src/lib/serviceRoutines/postQueryContext.cpp
@@ -531,13 +531,17 @@ std::string postQueryContext
   // In API version 2, this has changed completely. Here, the total count of local entities is returned
   // if the URI parameter 'count' is set to 'true', and it is returned in the HTTP header Fiware-Total-Count.
   //
-  std::string georel  =  ciP->uriParam["georel"];
+  /*std::string georel  =  ciP->uriParam["georel"];
   if ((ciP->apiVersion == V2))
   {
     if (!((ciP->inMimeType == JSON && strstr(ciP->payload, "near")) || strstr(georel.c_str(), "near")))
     {
       countP = &count;
     }
+  }*/
+  if ((ciP->apiVersion == V2) && (ciP->uriParamOptions["count"]))
+  {
+    countP = &count;
   }
   else if ((ciP->apiVersion == V1) && (ciP->uriParam["details"] == "on"))
   {
```

and it's seem it's working:

```
$ ./testHarness.sh cases/4114_geojson_feature_support
lun 05 jun 2023 10:00:11 CEST
Run tests 0 to 0
0001/6: geojson_feature_errors.test ..................................................................................  04 seconds
0002/6: geojson_feature_errors_ignoretype.test .......................................................................  04 seconds
0003/6: geojson_feature_support.test .................................................................................  02 seconds
0004/6: geojson_feature_support_ignoretype.test ......................................................................  03 seconds
0005/6: geojson_featurecollection_support.test .......................................................................  02 seconds
0006/6: geojson_featurecollection_support_ignoretype.test ............................................................  03 seconds
Total test time: 18.30 seconds

$ ./testHarness.sh cases/1038_ngsiv2_geo_location_entities/
lun 05 jun 2023 10:00:41 CEST
Run tests 0 to 0
0001/16: entity_location_shape_changes.test ...........................................................................  05 seconds
0002/16: errors_general.test ..........................................................................................  04 seconds
0003/16: errors_geobox.test ...........................................................................................  04 seconds
0004/16: errors_geoline.test ..........................................................................................  05 seconds
0005/16: errors_geopoint.test .........................................................................................  03 seconds
0006/16: errors_geopolygon.test .......................................................................................  04 seconds
0007/16: geoquery_box_geobox.test .....................................................................................  03 seconds
0008/16: geoquery_box_geobox_after_update.test ........................................................................  03 seconds
0009/16: geoquery_box_geoline.test ....................................................................................  03 seconds
0010/16: geoquery_box_geoline_after_update.test .......................................................................  03 seconds
0011/16: geoquery_box_geopolygon.test .................................................................................  02 seconds
0012/16: geoquery_box_geopolygon_after_update.test ....................................................................  03 seconds
0013/16: geoquery_circle_geopoint.test ................................................................................  05 seconds
0014/16: geoquery_circle_geopoint_after_update.test ...................................................................  04 seconds
0015/16: geoquery_polygon_geopoint.test ...............................................................................  06 seconds
0016/16: geoquery_polygon_geopoint_after_update.test ..................................................................  06 seconds
Total test time: 62.86 seconds

$ ./testHarness.sh cases/1177_geometry_and_coords/
lun 05 jun 2023 10:02:29 CEST
Run tests 0 to 0
0001/2: geometry_and_coords.test .....................................................................................  03 seconds
0002/2: geometry_and_coords_errors.test ..............................................................................  03 seconds
Total test time: 6.20 seconds
```